File: somatorio.dart
import 'dart:io';

int somatorioDivisiveis(int numero) {
  // Inicializa a variável de somatório.
  int somatorio = 0;

  // Loop que itera de 0 até (numero - 1).
  for (int i = 0; i < numero; i++) {
    // Verifica se o número atual (i) é divisível por 3 ou 5.
    if (i % 3 == 0 || i % 5 == 0) {
      // Se for divisível, adiciona ao somatório.
      somatorio += i;
    }
  }

// Retorna o resultado final.
  return somatorio;
}

void main() {
  // Solicita ao usuário que insira um número.
  stdout.write("Digite um número inteiro positivo: ");
  // Lê a entrada do usuário e converte para inteiro.
  int numero = int.parse(stdin.readLineSync()!);

  // Chama a função e armazena o resultado.
  int resultado = somatorioDivisiveis(numero);

  // Imprime o resultado.
  print(
      "O somatório dos valores divisíveis por 3 ou 5 abaixo de $numero é: $resultado");
}

File: somatorio.js
const readline = require('readline');

// Cria uma interface de leitura.
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

// Função que recebe um número inteiro positivo e retorna o somatório
// de todos os valores inteiros divisíveis por 3 ou 5 abaixo desse número.
function somatorioDivisiveis(numero) {
  let somatorio = 0;

  for (let i = 0; i < numero; i++) {
    if (i % 3 === 0 || i % 5 === 0) {
      somatorio += i;
    }
  }

  return somatorio;
}

// Solicita ao usuário que insira um número.
rl.question('Digite um número inteiro positivo: ', (numero) => {
  // Converte a entrada do usuário para inteiro.
  numero = parseInt(numero);

  // Chama a função e armazena o resultado.
  let resultado = somatorioDivisiveis(numero);

  // Imprime o resultado.
  console.log(`O somatório dos valores divisíveis por 3 ou 5 abaixo de ${numero} é: ${resultado}`);

  // Fecha a interface de leitura.
  rl.close();
});